src/lib.rs

```diff
@@ -65,12 +65,16 @@ pub trait UdpStack {
 	/// The type returned when we have an error
 	type Error: core::fmt::Debug;
 
-	/// Open a new UDP socket to the given address and port. UDP is connectionless,
-	/// so unlike `TcpStack` no `connect()` is required.
-	fn open(&self, remote: SocketAddr) -> Result<Self::UdpSocket, Self::Error>;
+	/// Open a UDP socket with a dynamically selected port.
```

Comment:
It's feasible someone may want to allocate a socket, but not connect it immediately. However, since this wasn't around in the previous implementation, I think we can ignore that use case and add it if needed later.